src/cyndilib/video_frame.pyx

```diff
@@ -202,6 +202,14 @@ cdef class VideoFrame:
         cdef bytes result = self.ptr.p_metadata
         return result
 
+    def set_metadata(self, bytes metadata):
+        """Set the video metadata. The lifetime of the metadata must be as long as the VideoFrame.
+        """
+        self._set_metadata(metadata)
+    cdef int _set_metadata(self, bytes metadata) noexcept:
```

Comment:
I'm not 100% certain that this can be marked as `noexcept` since we're dealing with Python objects here.

It'd be safer to mark is as `except -1`.